crates/hdk3/src/entry.rs

```diff
@@ -1,4 +1,597 @@
+/// reference other headers.
+/// Updates and deletes both reference a specific header+entry combination.
+/// Updates must reference another create or update header+entry.
+/// Deletes must reference a create or update header+entry (nothing can reference a delete).
+///
+/// Full elements are returned for direct references to the passed hash.
+/// Header hashes are returned for references to references to the passed hash.
+///
+/// [ `Details` ] for a header hash return:
+/// - the element for this header hash if it exists
+/// - all update and delete _elements_ that reference that specified header
+///
+/// [ `Details` ] for an entry hash return:
+/// - all creates, updates and delete _elements_ that reference that entry hash
+/// - all update and delete _elements_ that reference the elements that reference the entry hash
+///
+/// Note: Entries are just values, so can be referenced by many CRUD headers by many authors.
+///       e.g. the number 1 or string "foo" can be referenced by anyone publishing CRUD headers at
+///       any time they need to represent 1 or "foo" for a create, update or delete.
+///       If you need to disambiguate entry values, provide uniqueness in the entry value such as
+///       a unique hash (e.g. current chain head), timestamp (careful about collisions!), or random
+///       bytes/uuid (see random_bytes() and the uuid rust crate that supports uuids from bytes).
+///
+/// Note: There are multiple header types that exist and operate entirely outside of CRUD elements
+///       so they cannot reference or be referenced by CRUD, so are immutable or have their own
+///       mutation logic (e.g. link create/delete) and will not be included in [ `get_details` ] results
+///       e.g. the DNA itself, links, migrations, etc.
+///       However the element will still be returned by [ `get_details` ] if a header hash is passed,
+///       these header-only elements will have [ `None` ] as the entry value.
+pub fn get_details<H: Into<AnyDhtHash>>(
+    hash: H,
+    options: GetOptions,
+) -> ExternResult<Option<Details>> {
+    host_call::<GetInput, Option<Details>>(__get_details, GetInput::new(hash.into(), options))
+}
+
+/// Trait for binding static [ `EntryDef` ] property access for a type.
+/// See [ `register_entry` ]
+pub trait EntryDefRegistration {
+    fn entry_def() -> crate::prelude::EntryDef;
+
+    fn entry_def_id() -> crate::prelude::EntryDefId;
+
+    fn entry_visibility() -> crate::prelude::EntryVisibility;
+
+    fn crdt_type() -> crate::prelude::CrdtType;
+
+    fn required_validations() -> crate::prelude::RequiredValidations;
+}
+
+/// Implements conversion traits to allow a struct to be handled as an app entry.
+/// If you have some need to implement custom serialization logic or metadata injection
+/// you can do so by implementing these traits manually instead.
+///
+/// This requires that TryFrom and TryInto [ `derive@SerializedBytes` ] is implemented for the entry type,
```

Comment:
Is it true that we require `SerializedBytes`, or merely `Serialize + Deserialize`?

Reply:
@maackle true, i think now you can work with entries as `Serialize + Debug + Deserialize` actually

Reply:
@maackle actually it looks like app entry is still defined in terms of `SerializedBytes`